ref(apple): dSYMs restructuring (#6015)
getsentry/sentry-docs: src/platform-includes/debug-symbols-apple/_default.mdx

Every solution requires a Sentry Auth Token with the [correct permissions](/product/cli/dif/#permissions). You can [create one here](https://sentry.io/api/). To view uploaded dSYMs in your project, go to **[Project] > Settings > Debug Files**.
Upload the dSYMs using either:

* [sentry-cli](#sentry-cli)
* [Sentry Fastlane Plugin](#fastlane)
* [Xcode Build Phase](#xcode-build-phase)
* [App Store Connect Integration](#bitcode-appstore)

<Note>

Starting with Xcode 13, newly generated projects do not automatically create a required `Info.plist` configuration file.
If this is your situation, please read [Info.plist Is Missing in Xcode 13 — Here's How To Get It Back](https://betterprogramming.pub/info-plist-is-missing-in-xcode-13-heres-how-to-get-it-back-1a7abf3e2514).

</Note>

## sentry-cli

Use the `sentry-cli upload-dif` command to upload dSYMs to Sentry. The command will recursively scan the provided folders or ZIP archives. Files that have already been upload are skipped automatically.

> We recommend setting the `--include_sources` option to include sources for <PlatformLink to="/data-management/debug-files/source-context/">source context</PlatformLink> to work.

Author: When using the sentry-cli you only have to add `--include_sources` , when using Fastlane you have to set `include_sources` to `true`, if you set it to `false` it doesn't work.

</Alert>

## Xcode Build Phase

Your project's dSYM can be uploaded during Xcode's build phase as a _Run Script_. For this you need to set the _DEBUG_INFORMATION_FORMAT_ to be _DWARF with dSYM File_. By default, an Xcode project will only have _DEBUG_INFORMATION_FORMAT_ set to _DWARF with dSYM File_ in _Release_ so make sure everything is set in your build settings properly. To upload the debug symbols, you need to do the following:

1.  Download and install [sentry-cli](/product/cli/installation/)
2.  You will need to copy the script below into a new _Run Script_ and set your _AUTH_TOKEN_, _ORG_SLUG_, and _PROJECT_SLUG_.

> 3.  We recommend setting the `--include_sources` option to include sources for <PlatformLink to="/data-management/debug-files/source-context/">source context</PlatformLink> to work. Simply change `sentry-cli upload-dif` to `sentry-cli upload-dif --include_sources` in the copied script.

Author: > For source context to work, we recommend setting the include_sources parameter to true:"

No, it should state only adding `--include_sources`.

#### `Issue Details`

![Warning to update credentials in an issue's details](issue-details-warning.png)

#### `Project Settings`

![Warning to update credentials in sidebar](sidebar-warning.png)

#### `Custom Repositories`

![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)

When using Fastlane's _upload_to_testflight_ action, it is recommended to pass the _skip_waiting_for_build_processing_ parameter to speed up the action.

## Bitcode {#dsym-with-bitcode}

If [Bitcode](https://developer.apple.com/library/ios/documentation/IDEs/Conceptual/AppDistributionGuide/AppThinning/AppThinning.html#//apple_ref/doc/uid/TP40012582-CH35-SW2), no longer available [starting with Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes), **is enabled** in your project, you need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry.

> You can achieve this by using either:

Author: Almost, I changed the second Xcode 14 to Xcode 14.

> - Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
> - [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS direclty from App Store Connect.
> - Manually from App Store Connect:
>   - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
>   - Login to App Store Connect, go to your app, go to “Activity", click the build number to go into the detail page, and click _Download dSYM_.
>   - Manually upload the dSYMs with [sentry-cli](#sentry-cli).

Reviewer: ```suggestion
- Download Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
- Use the [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS directly from App Store Connect.
- Download manually from App Store Connect:
  - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
  - Log in to App Store Connect, go to your app > “Activity".
  - Click the build number to go into the detail page, and click _Download dSYM_.
  - Manually upload the dSYMs with [sentry-cli](#sentry-cli).
```

> These dSYMs contain obfuscated symbol names and paths. You must upload the BCSymbolMap file stored in your xcarchive so Sentry can  properly symbolicate your crash reports. You can do this by using the upload-dif command of either [sentry-cli](#sentry-cli) or the [Sentry Fastlane plugin](#fastlane), which will automatically find and upload all BCSymbolMap files, given the path to the app's xcarchive.

Reviewer: ```suggestion
These dSYMs contain obfuscated symbol names and paths. You must upload the BCSymbolMap file stored in your xcarchive so Sentry can properly symbolicate your crash reports. You can do this by using the upload-dif command of either [sentry-cli](#sentry-cli) or the [Sentry Fastlane plugin](#fastlane), which will automatically find and upload all BCSymbolMap files once it has the path to the app's xcarchive.
```
Want to make sure that the update I made to the last line makes sense? Am I understanding what you meant by "given the path to the app's...?" correctly?

Author: I tried it differently. Please have another look.

```

</Alert>

> ## App Store Connect {#appstore-connect}

Reviewer: ```suggestion
## App Store Connect {#appstore-connect} Integration
```

Setting up an App Store Connect custom repository requires an App Store Connect API Key. Follow the documentation on [Creating API Keys for App Store Connect API](https://developer.apple.com/documentation/appstoreconnectapi/creating_api_keys_for_app_store_connect_api) and provide the _Issuer_, _Key ID_, and _Private Key_. In order to be able to fetch build details, your API Key needs to be assigned the _Developer_ role.

**Invalid Credentials**

If your App Store Connect API key is revoked on the Apple side, the Sentry custom repository using this key's credentials will no longer be able to discover and fetch dSYM files. You'll see the following warnings in Sentry's user interface:

#### `Issue Details`

![Warning to update credentials in an issue's details](issue-details-warning.png)

#### `Project Settings`

![Warning to update credentials in sidebar](sidebar-warning.png)

#### `Custom Repositories`

![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)

When using Fastlane's _upload_to_testflight_ action, it's recommended to pass the _skip_waiting_for_build_processing_ parameter to speed up the action.

## Bitcode {#dsym-with-bitcode}

> The App Store will not accept bitcode submissions from [Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). If you’re using Xcode 13, you’ll need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry using one of the methods below:

Reviewer: ```suggestion
The App Store will not accept bitcode submissions from [Xcode 14](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes). If you’re using Xcode 14, you’ll need to download the dSYMs from App Store Connect after it finishes processing your build, and then upload them to Sentry using one of the methods below:
```

Should it be "if you're using Xcode 14?"

Author: No, it should be 13. If you use Xcode 14, it doesn't work.

- Download Fastlane's [download_dsyms](https://docs.fastlane.tools/actions/download_dsyms/) action and then upload the dSYMs with the [Sentry Fastlane plugin](#fastlane).
- Use the [Sentry App Store Connect integration](#appstore-connect), which fetches the dSYMS directly from App Store Connect.
- Download manually from App Store Connect:
  - Open Xcode Organizer, go to your app, and click _Download dSYMs..._
  - Log in to App Store Connect, go to your app > “Activity".

>   - Click the build number to go into the detail page, and click _Download dSYM_.

Reviewer: ```suggestion
  - Click the build number to go into the "detail" page, and click _Download dSYM_.
```
